src/credential/mod.rs

```diff
@@ -3,27 +3,258 @@ pub mod jwt_vc;
+            // then the assumption is that if there are any failed validations, then the credential does
+            // does not match the presentation definition requirements.
+            .any(|validation| !validation);
+
+        // Return the double-negative of the failed validation status.
+        Ok(!failed_validation)
+    }
+
+    /// Convert the parsed credential into a specialized JSON credential.
+    pub fn try_into_parsed(&self) -> Result<Arc<ParsedCredential>, CredentialDecodingError> {
+        self.to_owned().try_into()
+    }
+
+    /// Returns the credential as a `Mdoc` credential.
+    pub fn try_into_mdoc(&self) -> Result<Arc<Mdoc>, CredentialDecodingError> {
+        self.to_owned()
+            .try_into()
+            .map_err(|e: MdocInitError| CredentialDecodingError::MsoMdoc(e.to_string()))
+    }
+
+    /// Returns the credential as a `JwtVc` credential.
+    pub fn try_into_jwt_vc(&self) -> Result<Arc<JwtVc>, CredentialDecodingError> {
+        self.to_owned()
+            .try_into()
+            .map_err(|e: JwtVcInitError| CredentialDecodingError::JwtVc(e.to_string()))
+    }
+
+    /// Return the credential as an AnyJsonCredential type.
+    pub fn try_into_any_json_credential(
+        &self,
+    ) -> Result<AnyJsonCredential, CredentialDecodingError> {
+        self.to_owned().try_into()
+    }
+}
+
+impl TryFrom<Credential> for AnyJsonCredential {
+    type Error = CredentialDecodingError;
+
+    fn try_from(value: Credential) -> Result<Self, Self::Error> {
+        match value.format {
+            // NOTE: Re-using the Arc<JwtVc> type for SdJwtVcJson format
+            // for converting into the AnyJsonCredential type.
+            //
+            // There may be a better solution for this.
+            CredentialFormat::JwtVcJson | CredentialFormat::SdJwtVcJson => {
+                let jwt_vc: Arc<JwtVc> = value
+                    .try_into()
+                    .map_err(|e: JwtVcInitError| CredentialDecodingError::JwtVc(e.to_string()))?;
+                Ok(jwt_vc.credential())
+            }
+            // TODO: Add more formats here.
+            _ => Err(Self::Error::UnsupportedCredentialFormat(
+                value.format.to_string(),
+            )),
+        }
+    }
+}
+
+impl TryFrom<&Credential> for Value {
+    type Error = CredentialDecodingError;
+
+    fn try_from(credential: &Credential) -> Result<Self, Self::Error> {
+        match &credential.format {
+            CredentialFormat::JwtVcJson => {
+                let jwt_vc: Arc<JwtVc> = credential
+                    .try_into()
+                    .map_err(|e: JwtVcInitError| CredentialDecodingError::JwtVc(e.to_string()))?;
+                let value: Value = serde_json::to_value(jwt_vc.credential())
+                    .map_err(|e| CredentialDecodingError::Serialization(e.to_string()))?;
+                Ok(value)
+            }
+            CredentialFormat::SdJwtVcJsonLd => {
```

Comment:
```suggestion
            CredentialFormat::SdJwtVc => {
```

Is this the correct format?

Reply:
Also need to handle `CredentialFormat::SdJwtVcJson`.